tensorboard/webapp/widgets/data_table/filter_dialog_test.ts

```diff
@@ -0,0 +1,143 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {TestBed} from '@angular/core/testing';
+import {HarnessLoader} from '@angular/cdk/testing';
+import {TestbedHarnessEnvironment} from '@angular/cdk/testing/testbed';
+import {DataTableModule} from './data_table_module';
+import {FilterDialog} from './filter_dialog_component';
+import {DiscreteFilter, DomainType, IntervalFilter} from './types';
+import {By} from '@angular/platform-browser';
+import {RangeInputComponent} from '../range_input/range_input_component';
+import {RangeInputModule} from '../range_input/range_input_module';
+import {MatCheckboxModule} from '@angular/material/checkbox';
+import {MatCheckboxHarness} from '@angular/material/checkbox/testing';
+
+describe('filter dialog', () => {
+  let rootLoader: HarnessLoader;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [FilterDialog],
+      imports: [DataTableModule, RangeInputModule, MatCheckboxModule],
+    }).compileComponents();
+  });
+
+  function createComponent(input: {filter: DiscreteFilter | IntervalFilter}) {
+    const fixture = TestBed.createComponent(FilterDialog);
+    fixture.componentInstance.filter = input.filter;
+    rootLoader = TestbedHarnessEnvironment.documentRootLoader(fixture);
+    fixture.detectChanges();
+    return fixture;
+  }
+
+  it('dispatches event when an interval filter value is changed', async () => {
+    const fixture = createComponent({
+      filter: {
+        type: DomainType.INTERVAL,
+        includeUndefined: false,
+        minValue: 6,
+        maxValue: 20,
+        filterLowerValue: 7,
+        filterUpperValue: 18,
+      },
+    });
+
+    const intervalFilterChangedSpy = spyOn(
+      fixture.componentInstance.intervalFilterChanged,
+      'emit'
+    );
+
+    const rangeInput = fixture.debugElement.query(
+      By.directive(RangeInputComponent)
+    );
+    rangeInput.componentInstance.rangeValuesChanged.emit({
+      lowerValue: 7,
+      upperValue: 17,
+    });
+    expect(intervalFilterChangedSpy).toHaveBeenCalled();
+
+    const includeUndefinedSpy = spyOn(
+      fixture.componentInstance.includeUndefinedToggled,
+      'emit'
+    );
+
+    const includeUndefinedCheckbox = await rootLoader.getHarness(
+      MatCheckboxHarness.with({label: 'Include Undefined'})
+    );
+    await includeUndefinedCheckbox.check();
+    expect(includeUndefinedSpy).toHaveBeenCalled();
+  });
+
+  it('dispatches event when an discrete filter value is changed', async () => {
+    const fixture = createComponent({
+      filter: {
+        type: DomainType.DISCRETE,
+        includeUndefined: false,
+        possibleValues: [2, 4, 6, 8],
+        filterValues: [2, 4, 6],
```

Comment:
Can we add coverage of what is rendered? Given possibleValues and filterValues, which checkboxes are rendered? (Ideally in its own test case).